cMake/FreeCAD_Helpers/FindSalomeMESH.cmake

```diff
@@ -0,0 +1,126 @@
+macro(FindSalomeMESH)
+# -------------------------------- Salome SMESH --------------------------
+
+    # Salome SMESH sources are under src/3rdParty now
+    if(BUILD_SMESH)
+        # set the internal smesh version:
+        # see src/3rdParty/salomonemesh/CMakeLists.txt and commit https://github.com/FreeCAD/FreeCAD/commit/666a3e5 and https://forum.freecadweb.org/viewtopic.php?f=10&t=30838
```

Comment:
s/salomonemsh/salomesmesh/

Reply:
Good catch, I made this change locally but failed to add it to the git repo.

Reply:
The "salomonemsh" error is [in master](https://github.com/FreeCAD/FreeCAD/blob/master/CMakeLists.txt#L680). For this initial PR, I wanted to literally copy/paste what is currently in master and simply move it to the cMake sub-directory. In a subsequent future PR I will start going through the actual code/logic and try to clean things like this up.